rest/oidc_api_test.go

```diff
@@ -547,7 +547,7 @@ func TestOpenIDConnectAuth(t *testing.T) {
 			authURL:         "/db/_oidc?provider=foo&offline=true",
 			forceAuthError: forceError{
 				errorType:            callbackTokenExchangeErr,
-				expectedErrorCode:    http.StatusInternalServerError,
+				expectedErrorCode:    http.StatusUnauthorized,
```

Comment:
Do these represent a change in behaviour between the previous and current OIDC implementations?  Can you remind me of the rationale for this change?  Same applies to the two scenarios below.

Reply:
Per https://openid.net/specs/openid-connect-core-1_0.html#RefreshErrorResponse, on error it's going to return these errors: https://tools.ietf.org/html/rfc6749#section-5.2.  That looks like we'd be able to differentiate between invalid requests and failed authentications.

We should only be returning a 401 for failed authentications - any unexpected errors should continue to be treated as 500.

Reply:
Make sense, thanks Adam for clarifying this. Revised in latest revision.